ibm_mq/datadog_checks/ibm_mq/collectors/queue_metric_collector.py

```diff
@@ -94,6 +95,12 @@ def _discover_queues(self, queue_manager, mq_pattern_filter):
                 for queue_info in response:
                     queue = queue_info[pymqi.CMQC.MQCA_Q_NAME]
                     queues.append(to_string(queue).strip())
+            finally:
+                # Close internal reply queue to prevent filling up a dead-letter queue.
+                # https://github.com/dsuch/pymqi/blob/084ab0b2638f9d27303a2844badc76635c4ad6de/code/pymqi/__init__.py#L2892-L2902
+                # https://dsuch.github.io/pymqi/examples.html#how-to-specify-dynamic-reply-to-queues
+                if pcf is not None:
+                    pcf.disconnect()
```

Comment:
any chances this can throw an exception?

Reply:
It shouldn't as exceptions are handled inside this method
https://github.com/dsuch/pymqi/blob/35d0e45ea16c0b2077b20906856f587d6f8ffa6a/code/pymqi/__init__.py#L2900-L2910